test/expect-tests/stdune/ansi_color_tests.ml

```diff
@@ -0,0 +1,482 @@
+open Stdune
+
+let dyn_of_pp tag pp =
+  let open Dyn.Encoder in
+  let rec conv = function
+    | Pp.Ast.Nop -> constr "Nop" []
+    | Seq (x, y) -> constr "Seq" [ conv x; conv y ]
+    | Concat (x, y) -> constr "Concat" [ conv x; list conv y ]
+    | Box (i, x) -> constr "Box" [ int i; conv x ]
+    | Vbox (i, x) -> constr "Vbox" [ int i; conv x ]
+    | Hbox x -> constr "Hbox" [ conv x ]
+    | Hvbox (i, x) -> constr "Hvbox" [ int i; conv x ]
+    | Hovbox (i, x) -> constr "Hovbox" [ int i; conv x ]
+    | Verbatim s -> constr "Verbatim" [ string s ]
+    | Char c -> constr "Char" [ char c ]
+    | Break (x, y) ->
+      let f = triple string int string in
+      constr "Break" [ f x; f y ]
+    | Newline -> constr "Newline" []
+    | Tag (ta, t) -> constr "Tag" [ tag ta; conv t ]
+    | Text s -> constr "Text" [ string s ]
+  in
+  conv (Pp.to_ast pp)
+
+let%expect_test "reproduce #2664" =
+  (* https://github.com/ocaml/dune/issues/2664 *)
+  let b = Buffer.create 100 in
+  let f s = Buffer.add_string b ("\027[34m" ^ s ^ "\027[39m") in
+  for i = 1 to 20 do
+    f (string_of_int i)
+  done;
+  let pp =
+    Buffer.contents b |> Ansi_color.parse
+    |> dyn_of_pp (Dyn.Encoder.list Ansi_color.Style.to_dyn)
+    |> Dyn.pp
+  in
+  Format.printf "%a@.%!" Pp.to_fmt pp;
+  [%expect {|
```

Comment:
Nice that we can see the representation of `Pp.t` :)

I would also print the original input together with the rendered one, so that we can see the full effect of the "ansi -> Pp.t -> ansi" round trip.

Reply:
Done